src/basic/strv.c

```diff
@@ -789,6 +789,22 @@ int strv_extend_n(char ***l, const char *value, size_t n) {
         return -ENOMEM;
 }
 
+int strv_extend_join(char ***l, const char *lhs, const char *rhs) {
+        char *j;
+
+        assert(l);
+        assert(lhs);
+
+        if (!rhs) /* value is optional, in which case we suppress the field */
+                return 0;
```

Comment:
hmm, these semantics sound quite specific to me. should this really be in strv?

at the very least it should have a different name, i.e. strv_extend_with_assignment() or so?

shouldn't the `=` be inserted automatically here?